### ISSUE TITLE
功能请求：在 Crater 系统中集成 TensorBoard 可视化支持

### ISSUE THREAD

**Issue by Jinghao-coding** [raids-lab/crater#173]
### Feature Description

**概要**
希望在 Crater 系统中集成 **TensorBoard**，用于统一展示训练过程、性能分析以及实验结果的可视化界面。

---

#### 🧠 背景与动机

当前 Crater 已具备实验管理与性能监控功能，但缺少统一的实时可视化入口来展示：

* 模型训练指标（如 loss、accuracy、learning rate 等）；
* 性能分析结果（如算子级延迟、GPU 利用率等）；
* 多实验结果对比与可视化。

集成 TensorBoard 将大幅提升系统的 **易用性与科研可视化能力**，方便研究者和开发者快速分析训练效果与性能瓶颈。

---

#### 🚀 建议方案

在 Crater 中提供 **可选的 TensorBoard 集成模块**，具体包括：

1. **日志输出支持**

   * 在训练和推理过程中，自动使用 `SummaryWriter` 输出指标与性能数据；
   * 支持分层命名（如按模型、数据集或配置 ID 区分不同实验）。

2. **命令行启动接口**

   * 新增命令：

     ```bash
     crater tensorboard --logdir /path/to/logs
     ```
   * 在系统前端或 Web 控制台中，提供一键跳转或嵌入式查看入口。

3. **分布式与集群环境支持**

   * 支持聚合多个节点的日志输出；
   * 允许通过端口转发或代理访问远程 TensorBoard 服务。

---

#### 📊 预期收益

* 提供统一的训练指标与系统性能可视化；
* 方便对不同实验结果进行对比分析；
* 提升科研复现与性能优化的可观测性。

---

#### ⚙️ 可选扩展功能

* 将 Crater 的性能分析结果转化为 TensorBoard 可读取的事件文件（`events.out.tfevents`）；
* 支持通过配置文件开启功能，例如：

  ```yaml
  monitor:
    tensorboard: true
  ```
* 在未来支持 TensorBoard 的 trace viewer，与 Crater 的 profiler 输出对接。

---

#### 📚 参考资料

* [[TensorBoard: Visualizing Learning](https://www.tensorflow.org/tensorboard)](https://www.tensorflow.org/tensorboard)
* [[PyTorch SummaryWriter API](https://pytorch.org/docs/stable/tensorboard.html)](https://pytorch.org/docs/stable/tensorboard.html)
* 类似工具参考：MLflow、Weights & Biases (WandB)
